lib/spack/spack/environment/environment.py

```diff
@@ -3059,6 +3059,34 @@ def use_config(self):
             self.deactivate_config_scope()
 
 
+def add_path_scopes(cfg: spack.config.Configuration, name: str, path: str):
+    """Add configuration scope(s) from the path
+
+    Arguments:
+        cfg: configuration instance
+        name: configuration scope name
+        path: path to configuration file(s)
+
+    Raises:
+        spack.error.ConfigError: if the path is an invalid configuration scope
+    """
+    if exists(path):  # managed environment
+        manifest = EnvironmentManifestFile(root(path))
+    elif is_env_dir(path):  # anonymous environment
+        manifest = EnvironmentManifestFile(path)
+    elif os.path.isdir(path):  # directory with config files
+        cfg.push_scope(spack.config.DirectoryConfigScope(name, path, writable=False))
+        spack.config._add_platform_scope(cfg, name, path, writable=False)
+        return
```

Comment:
This bit feels wrong for `spack.environment`

Instead of taking `cfg` and modifying it, this function could return `Optional[List[spack.config.ConfigScope]]`. So, if the relevant path is an environment, then return the relevant config scopes, else return `None`.

Then the call site can have this `elif os.path.isdir(...)` branch and add `DirectoryConfigScope`s instead.

Reply:
I agree and will revisit. Thanks for the suggestion.

Reply:
Done.